core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -504,7 +508,8 @@ UNWIND(	.cantunwind)
 
 #if defined (CFG_BOOT_SECONDARY_REQUEST)
 	/* if L1 is not invalidated before, do it here */
-	bl	arm_cl1_d_invbysetway
+	mov	r0, #DCACHE_OP_INV
+	bl	dcache_op_louis
```

Comment:
Sorry, late comment after reading back this.
Actually, i even think this should invalidate only the L1 (`dcache_op_level1`), not the L2 (in case it is in the inner) that might be used by other running cores.